proposals/primer/application.bs

```diff
@@ -0,0 +1,353 @@
+<pre class='metadata'>
+Title: Solid Application Interoperability - Application Primer
+Shortname: solid-interop-application-primer
+Level: 1
+Status: w3c/ED
+Group: solid-cg
+URL: https://solid.github.io/data-interoperability-panel/primer/applications.html
+Repository: https://github.com/solid/data-interoperability-panel
+Inline Github Issues: title
+Editor: elf Pavlik
+Abstract: Primer for libraries intented to be used by applications.
+Boilerplate: omit conformance
+</pre>
+
+# Introduction # {#introduction}
+
+This primer was developed along side open source TypeScript implementation [[SAI-JS]]
+
+# Shape Trees # {#shape-trees}
+
+Shape Trees [[SHAPETREES]] allow creating virtual data hierarchies and define which shapes [[SHEX]]
+should be used to validate data. In our examples we will use following Shapes and Shape Trees.
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/trees/Project.ttl
+  highlight: turtle
+  show: 6-100
+  </pre>
+  <figcaption>Project shape tree</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/trees/Task.ttl
+  highlight: turtle
+  show: 5-100
+  </pre>
+  <figcaption>Task shape tree</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/shapes/Project.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Project shape (ShEx)</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/shapes/Task.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Task shape (ShEx)</figcaption>
+</figure>
+
+Issue: Reference shape path documentation
+
+# Application # {#appliction}
+
+In this primer we will consider case of an application
+denoted by <code>https://projectron.example/#app</code>
+Every application has public WebID Document providing information about it.
+
+
+
+# Authorization Flow # {#authorization-flow}
+
+User always needs to authenticate first.
+For the rest of this primer we'll be assuming that user has already authenticated,
+and that applicaction know WebID of the authenticated user.
+
+## Authorization Agent Discovery ## {#authorization-agent-discovery}
+
+Every user has an Authorization Agent which can be discovered from
+their WebID Document via <code>interop:hasAuthorizationAgent</code> predicate.
+
+<figure>
+  <pre class=include-code>
+  path: snippets/alice.example/alice.example.ttl
+  highlight: turtle
+  show: 3-100
+  line-start: 1
+  line-highlight: 3
+  </pre>
+  <figcaption>Alice's WebID document</figcaption>
+</figure>
+
+Here we see that Alice designates <code>https://auth.alice.example/</code> as their Authorization Agent.
+
+## Application Registration Discovery ## {#application-registration-discovery}
+
```

Comment:
Should be able to link to the operation included in #169 in the API document once that is merged

Reply:
added on line 143